vaadin-button-flow-parent/vaadin-button-flow/src/main/java/com/vaadin/flow/component/button/Button.java

```diff
@@ -359,10 +362,15 @@ public boolean isDisableOnClick() {
      * if server-side handling takes some time.
      */
     private void initDisableOnClick() {
-        getElement().executeJs("var disableEvent = function () {"
-                + "if($0.getAttribute('disableOnClick')){"
-                + " $0.setAttribute('disabled', 'true');" + "}" + "};"
-                + "$0.addEventListener('click', disableEvent)");
+        if(!disableInited) {
```

Comment:
Let's not add the `disableInited` flag since it will make the feature not work on detach -> attach / preserve on refresh.

Better to do the check in the JS function:

```js
if (!button.__hasDisableOnClickListener) {
  button.addEventListener('click', disableOnClickListener);
  button.__hasDisableOnClickListener = true;
}
```

Reply:
done